src/Symfony/Component/Routing/Matcher/UrlMatcher.php

```diff
@@ -259,4 +262,28 @@ protected function createRequest($pathinfo)
             'SCRIPT_NAME' => $this->context->getBaseUrl(),
         ));
     }
+
+    /**
+     * Match all variables enclosed in "{}" and substitute value using related
+     * attribute key.
+     *
+     * @param array $defaults
+     * @param array $attributes
+     */
+    private function resolveDefaults(&$defaults, $attributes)
+    {
+        $enclosingPattern = "/{(\w+)}/";
+
+        foreach ($defaults as $key => $value) {
+            $valueProcessed = preg_replace_callback($enclosingPattern, function ($match) use ($attributes) {
+                if (!isset($attributes[$match[1]])) {
```

Comment:
That will throw a logic exception: "Default routes with dynamic values should has a value."